specification/_types/common.ts

```diff
@@ -238,9 +228,13 @@ export enum OpType {
   create = 1
 }
 
-export type Refresh = boolean | RefreshOptions
-export enum RefreshOptions {
-  wait_for = 1
+// ES uses a string but accepts a json boolean for strings
+export type Refresh = boolean | RefreshValues
```

Comment:
Isn't the type definition of `Refresh` redundant, because boolean values are implied by `RefreshValues`? Which means we could just define `Refresh = RefreshValues` or directly as enum.

Reply:
Very good point. The "problem" is that in dynamically typed languages people may expect to just set this to `true` (the boolean value) instead of `"true"` (the string). Elasticsearch uses the string version, but its lenient JSON parsing transparently converts a JSON boolean to a string.

In Java this union makes things unnecessarily complex, so in the model preparation phase we rename the `RefreshValues` type to `Refresh` and only expose the enum, and same for `DynamicMapping`.

Note though that we could also handle this the other way around: if an enum contains `true` and `false`, then dynamically typed languages could generate the union by themselves. I'll do this in the TS generator and simplify the spec.

Reply:
Done in a62542c! And it also caught the same pattern for `IndicesIndexCheckOnStartup`. Code generation FTW 😄

Reply:
Awesome! Thanks for the swift turnaround and explanation.